lib/helpers/clone.js

```diff
@@ -145,4 +143,11 @@ function cloneArray(arr, options) {
   }
 
   return ret;
-}
\ No newline at end of file
+}
+
+function cloneRegExp(regexp) {
+  const ret = new RegExp(regexp);
+
+  (typeof regexp.lastIndex === 'number') && (ret.lastIndex = regexp.lastIndex);
```

Comment:
I think it'd be easier to read if we convert this to a regular condition.

Reply:
```suggestion
  If (typeof regexp.lastIndex === 'number') {
    ret.lastIndex = regexp.lastIndex;
  }
```